packages/api/lambdas/write-db-dlq-records-to-s3.js

```diff
@@ -8,7 +10,99 @@ const { s3PutObject } = require('@cumulus/aws-client/S3');
 const { parseSQSMessageBody } = require('@cumulus/aws-client/SQS');
 const { getMessageExecutionName } = require('@cumulus/message/Executions');
 const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
+const { getCumulusMessageFromExecutionEvent } = require('@cumulus/message/StepFunctions');
+
+/**
+ *
+ * @typedef {import('@cumulus/types').MessageGranule} MessageGranule
+ * @typedef {{granules: Array<MessageGranule>}} PayloadWithGranules
+*/
+
+/**
+ * @param {unknown} payload
+ * @returns {payload is PayloadWithGranules}
+ */
+function payloadHasGranules(payload) {
+  return (
+    payload instanceof Object
+    && 'granules' in payload
+    && Array.isArray(payload.granules)
+  );
+}
+/**
+ *
+ * @typedef {import('aws-lambda').EventBridgeEvent} EventBridgeEvent
+*/
+/**
+ * @param {{ [key: string]: any }} event
+ * @returns {event is EventBridgeEvent}
+ */
+const isEventBridgeLike = (event) => {
+  if (!(event instanceof Object && 'detail' in event && event.detail instanceof Object)) {
+    return false;
+  }
+  if (event.detail.status === 'RUNNING' && !('input' in event.detail)) {
+    return false;
+  }
+  if (event.detail.status === 'SUCCEEDED' && !('output' in event.detail)) {
+    return false;
+  }
+  return true;
+};
 
+/**
+ * Reformat object with key attributes at top level.
+ *
+ * @param {{ [key: string]: any }} messageBody - event bridge event as defined in aws-lambda
+ * @returns {Promise<Object>} - message packaged with metadata or 'unknown' where metadata not found
```

Comment:
With the correct typing, this is no longer correct.  Either fix, or remove/let automagic typing handle it.